cart/domain/cart/paymentselection.go

```diff
@@ -24,6 +26,8 @@ type (
 		ItemSplit() PaymentSplitByItem
 		TotalValue() price.Price
 		MethodByType(string) string
+		IdempotencyKey() string
+		GenerateNewIdempotencyKey() PaymentSelection
```

Comment:
maybe change to `GenerateNewIdempotencyKey() PaymentSelection,error` to properly handle errors in Idempotency Key generation.